### PR TITLE
Update debug info file names

### PR DESCRIPTION
**Pull request by keithc-ca**
* use `.debuginfo` on most platforms
* use `.dSYM` bundle on MacOSX
* continue using `.pdb` on Windows

(first commit tidies up whitespace and makes indentation consistent)

### CODE REVIEW

File: omrmakefiles/rules.aix.mk [hunk 1]
```diff
@@ -104,61 +103,58 @@ endif
...
-endef
-
-define LINK_CXX_SHARED_COMMAND
--$(RM) $@
-$(CXXLINKSHARED) -o $@ $(OBJECTS) $(LDFLAGS) $(MODULE_LDFLAGS) $(GLOBAL_LDFLAGS)
-cp $@ $@.dbg
-endef
 
+  # Add map files to clean target
+  define CLEAN_COMMAND
```
Comment (youngar) [inline omrmakefiles/rules.aix.mk]: I think if this has its body indented, the whitespace will be copied when the command is used. I'm not sure about this.

**Comment by youngar** [comment]
I like this change overall.  Some of the weird white space issues I think were due to old versions of make, or non-gnu make.  Was this tested on the IBM internal builds? I find they have the oldest versions of make.

File: omrmakefiles/rules.aix.mk (hunk 1 above)
Comment (keithc-ca) [reply omrmakefiles/rules.aix.mk]: I did a small test with make 4.1: no leading whitespace as commands are echoed.

**Comment by youngar** [comment]
@genie-omr build aix osx xlinux

**Comment by keithc-ca** [comment]
> Was this tested on the IBM internal builds?

I started a build; I'll send you a link privately.

**Comment by keithc-ca** [comment]
I plan to make cmake builds produce the same artifacts. I'll need to learn some of the finer details about how cmake works, so that will come later.

**Comment by youngar** [comment]
There might be an easy way, but you might have to add a `omr_add_library` macro.

**Pull request merged**